blueman/plugins/applet/TransferService.py

```diff
@@ -23,6 +25,33 @@
     from html import escape
 
 
+@GObject.type_register
+class CommandRunner(threading.Thread, GObject.Object):
```

Comment:
That's not correct.  `idle_add` will cause the execution of this code to freeze the Python interpreter in `subprocess.check_call` because that code will run on the main application thread.  That code needs to run asynchronously and in parallel to other tasks that the TransferService may be doing.

This is the correct solution to the problem — spawn a thread to monitor the process, reap the process (all done by `subprocess.check_call`) and then use `idle_add` to signal the return value of the process to the event handler subscribed to this class' outcome.